src/test/java/org/opensearch/ad/transport/SearchTopAnomalyResultActionTests.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ad.transport;
+
+import static org.opensearch.ad.settings.AnomalyDetectorSettings.BATCH_TASK_PIECE_INTERVAL_SECONDS;
+import static org.opensearch.ad.settings.AnomalyDetectorSettings.MAX_BATCH_TASK_PER_NODE;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+
+import org.junit.Before;
+import org.opensearch.ad.HistoricalAnalysisIntegTestCase;
+import org.opensearch.ad.TestHelpers;
+import org.opensearch.ad.model.AnomalyDetector;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import com.google.common.collect.ImmutableList;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST)
+public class SearchTopAnomalyResultActionTests extends HistoricalAnalysisIntegTestCase {
```

Comment:
Add a happy case also maybe?

Reply:
Can't add happy cases due to issues with the `lang-painless` module not being installed on test clusters brought up with `OpenSearchIntegTest`s, so I cover those at the rest-level. I actually had a comment about that in there originally, I will add it back to prevent confusion.

Reply:
See https://github.com/elastic/elasticsearch/issues/41109

Reply:
Okay I see, thanks for linking the issue. Comment could be nice to add as I just instinictaly thought of needing a happy case as I forgot about you mentioning this before.